### PR TITLE
Add support Rohm bh1750 ambient light sensor

### PR DESCRIPTION
**Pull request by jeronimoagullo**
Product Homepage:

https://www.rohm.com/

DataSheet:
https://www.farnell.com/datasheets/1813320.pdf

Testing Environment:
STMicroelectronics B_L4S5I_IOT01A Discovery kit

Signed-off-by: Jeronimo Agullo Ocampos [jeronimoagullo97@gmail.com](mailto:jeronimoagullo97@gmail.com)

### CODE REVIEW

File: drivers/sensor/bh1750/bh1750.c
Comment (teburd) [inline drivers/sensor/bh1750/bh1750.c:140]: SENSOR_DT_INST_DEFINE

File: drivers/sensor/bh1750/bh1750.c
Comment (teburd) [inline drivers/sensor/bh1750/bh1750.c:9]: The comments per include are probably a bit verbose

File: drivers/sensor/bh1750/bh1750.c
Comment (teburd) [inline drivers/sensor/bh1750/bh1750.c:67]: Are the delays coming from a datasheet describing them? If so please mention the section/page/value the datasheet mentions here for each delay.

**Review by teburd — CHANGES_REQUESTED** [review]

File: drivers/sensor/bh1750/bh1750.c
Comment (jeronimoagullo) [reply drivers/sensor/bh1750/bh1750.c:9]: I agree on it. This driver was my first approach to the Sensor API and I documented so much. I forgot to delete these verbose comments, I will remove them.

File: drivers/sensor/bh1750/bh1750.c
Comment (jeronimoagullo) [reply drivers/sensor/bh1750/bh1750.c:67]: Yes, it is described in the page 5 section "Measurement mode explanation", it is 120 ms for H_RESOLUTION_MODE and H_RESOLUTION_MODE2 and 16 ms for L_RESOLUTION_MODE. 

I propose the following comments:

```c
        // We need to wait for the new measurement to be taken
	// According to datasheet page 5 section "Measurement mode explanation"
	// 120 ms for H_RESOLUTION_MODE and H_RESOLUTION_MODE2
	// 16 ms for L_RESOLUTION_MODE
	#ifdef CONFIG_BH1750_H_RESOLUTION_MODE
		k_msleep(120);
```

File: drivers/sensor/bh1750/bh1750.c
Comment (jeronimoagullo) [reply drivers/sensor/bh1750/bh1750.c:140]: I think you mean `SENSOR_DEVICE_DT_INST_DEFINE`. 

I propose the following changes:
```suggestion
	SENSOR_DEVICE_DT_INST_DEFINE(inst, bh1750_init, NULL,			      \
```

**Comment by galak** [comment]
DT binding looks ok.

File: drivers/sensor/bh1750/Kconfig
Comment (MaureenHelm) [inline drivers/sensor/bh1750/Kconfig:7]: Don't need this anymore. Please remove

File: drivers/sensor/bh1750/Kconfig
Comment (MaureenHelm) [inline drivers/sensor/bh1750/Kconfig:48]: Please move to devicetree properties

File: drivers/sensor/bh1750/bh1750.c
Comment (MaureenHelm) [inline drivers/sensor/bh1750/bh1750.c:60]: Use `/* */` comments. Please fix throughout the driver

File: drivers/sensor/bh1750/bh1750.c
Comment (MaureenHelm) [inline drivers/sensor/bh1750/bh1750.c:99]:
```suggestion
static int bh1750_init(const struct device *dev)
```

File: drivers/sensor/bh1750/bh1750.c
Comment (MaureenHelm) [inline drivers/sensor/bh1750/bh1750.c:138]: Please check your editor settings (tab is 8 spaces)

File: dts/bindings/sensor/rohm,bh1750.yaml
Comment (MaureenHelm) [inline dts/bindings/sensor/rohm,bh1750.yaml:12]:
```suggestion
include: [sensor-device.yaml, i2c-device.yaml]
```

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thank you for the contribution!

Please add this sensor to `tests/drivers/build_all/sensor` to ensure it builds in CI, and also address requested changes by amending your comments and force pushing.

File: drivers/sensor/bh1750/Kconfig
Comment (jeronimoagullo) [reply drivers/sensor/bh1750/Kconfig:7]: Thank you for your kind words!

Ok, I delete it

File: drivers/sensor/bh1750/Kconfig
Comment (jeronimoagullo) [reply drivers/sensor/bh1750/Kconfig:48]: Just to be sure I do correctly. 
You mean to move to `dts/bindigns/sensor/rohm,bh1750.yaml` , don't you?
Thus, the sensor will not require any Kconfig variable and all settings would be configured by the device tree, right?
And just for curiosity, is it the new tendency of Zephyr development? 
Thank you in advance!

File: drivers/sensor/bh1750/bh1750.c
Comment (jeronimoagullo) [reply drivers/sensor/bh1750/bh1750.c:60]: Ok, done!

File: drivers/sensor/bh1750/bh1750.c
Comment (jeronimoagullo) [reply drivers/sensor/bh1750/bh1750.c:99]: Ok, done!

File: drivers/sensor/bh1750/bh1750.c
Comment (jeronimoagullo) [reply drivers/sensor/bh1750/bh1750.c:138]: Sorry, there were some spaces between tabs, I have checked it out and now it should be fine.

**Comment by MaureenHelm** [comment]
Please implement requested changes by amending the original commit rather than adding new commits. You also need to rebase to resolve merge conflicts.

**Comment by jeronimoagullo** [comment]
> Please implement requested changes by amending the original commit rather than adding new commits. You also need to rebase to resolve merge conflicts.

I have added the modifitions according to comments and then stashed all commits into the original commit. Then, I have merge, however, the merge has provoked a new commit. I am not sure if it is rigth, may you confirm it? Thank you in advance

File: dts/bindings/sensor/rohm,bh1750.yaml
Comment (jeronimoagullo) [reply dts/bindings/sensor/rohm,bh1750.yaml]: Done

File: drivers/sensor/bh1750/Kconfig
Comment (jeronimoagullo) [reply drivers/sensor/bh1750/Kconfig]: Ok, thank you very much for the clarification. 
Done

**Comment by MaureenHelm** [comment]
> > Please implement requested changes by amending the original commit rather than adding new commits. You also need to rebase to resolve merge conflicts.
> 
> I have added the modifitions according to comments and then stashed all commits into the original commit. Then, I have merge, however, the merge has provoked a new commit. I am not sure if it is rigth, may you confirm it? Thank you in advance

No, you need to rebase on the command line, not in the GH web interface, to remove the merge commit

**Comment by jeronimoagullo** [comment]
> > > Please implement requested changes by amending the original commit rather than adding new commits. You also need to rebase to resolve merge conflicts.
> > 
> > 
> > I have added the modifitions according to comments and then stashed all commits into the original commit. Then, I have merge, however, the merge has provoked a new commit. I am not sure if it is rigth, may you confirm it? Thank you in advance
> 
> No, you need to rebase on the command line, not in the GH web interface, to remove the merge commit

Ok, I understood. I think that it is now correct. It has passed all automated tests. Please, check it out. Thank you.

File: dts/bindings/sensor/rohm,bh1750.yaml
Comment (MaureenHelm) [inline dts/bindings/sensor/rohm,bh1750.yaml:22]: Please justify the chosen default value in the description. Is it a power on register reset value?

File: tests/drivers/build_all/sensor/i2c.dtsi
Comment (MaureenHelm) [inline tests/drivers/build_all/sensor/i2c.dtsi:653]: Please set the address to 0x55 to make it sequential. Note this is a build test that will never run on real hardware

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/bh1750/bh1750.c
Comment (jeronimoagullo) [reply drivers/sensor/bh1750/bh1750.c:115]: ok

File: drivers/sensor/bh1750/bh1750.c
Comment (jeronimoagullo) [reply drivers/sensor/bh1750/bh1750.c:118]: ok

File: tests/drivers/build_all/sensor/i2c.dtsi
Comment (jeronimoagullo) [reply tests/drivers/build_all/sensor/i2c.dtsi:653]: Ok, I see it. However, the previous one `test_i2c_as5600` is incorrect and should be `64` to follow the correct numbering. Thus, I propose to set the `test_i2c_bh1750` address to 0x65.

File: dts/bindings/sensor/rohm,bh1750.yaml
Comment (jeronimoagullo) [reply dts/bindings/sensor/rohm,bh1750.yaml:22]:
```suggestion
      Sensor resolution. 
      The values are the corresponding hexadecimal values to 
      write to configuration register.
      Default H-resolution mode (1lx) since it is the recommended
      mode in the page 5 of datasheet.
```

File: dts/bindings/sensor/rohm,bh1750.yaml
Comment (jeronimoagullo) [reply dts/bindings/sensor/rohm,bh1750.yaml:36]:
```suggestion
      Specifies the measurement mode used by the driver.
      The options are continously mode (one new measurement 
      is ready continously) or one-time-mode (perform only one
      measurement and then set sensor power down mode).
      Default CONTINOUSLY-MODE for continous sensing, more 
      common value. 
```
the sensor uses any of the modes as "power on" as it is describes in the figure of page 4 of the datasheet

File: drivers/sensor/bh1750/Kconfig
Comment (MaureenHelm) [inline drivers/sensor/bh1750/Kconfig:8]: Fix the indentation
```suggestion
	bool "BH1750 ambient light sensor"
	default y
```

File: drivers/sensor/bh1750/bh1750.h
Comment (MaureenHelm) [inline drivers/sensor/bh1750/bh1750.h:40]: I don't think it's worth having a separate private include file for this. Please move to `bh1750.c`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/bh1750/Kconfig
Comment (jeronimoagullo) [reply drivers/sensor/bh1750/Kconfig:8]: ok

File: drivers/sensor/bh1750/bh1750.h
Comment (jeronimoagullo) [reply drivers/sensor/bh1750/bh1750.h:40]: Ok, I think so too.

**Review by MaureenHelm — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]

**Comment by MaureenHelm** [comment]
> Hi @MaureenHelm and @teburd, it's some days since my latest changes and it passed all tests, I would like to know if something else is missing from my side. Thank you in advance.

We're currently in feature freeze and need to wait until the merge window reopens after the release.
https://github.com/zephyrproject-rtos/zephyr/wiki/Release-Management

**Comment by fabiobaltieri** [comment]
@jeronimoagullo can you rebase please?

**Comment by jeronimoagullo** [comment]
> @jeronimoagullo can you rebase please?

I have just rebased and pushed. I have merged my contribution with the commit of same sensor with which it was conflicting

**Comment by carlescufi** [comment]
@jeronimoagullo can you fix the compliance checks:

```
TRAILING_WHITESPACE: trailing whitespace
26
File:drivers/sensor/bh1750/bh1750.c
27
Line:132
28
 SPACING: space required after that ',' (ctx:VxV)
29
File:drivers/sensor/bh1750/bh1750.c
30
Line:138
31
 SPACING: space required after that ',' (ctx:VxV)
32
File:drivers/sensor/bh1750/bh1750.c
33
Line:138
34
 SPACING: space required after that ',' (ctx:VxV)
35
File:drivers/sensor/bh1750/bh1750.c
36
Line:139
37
 SPACING: space required after that ',' (ctx:VxV)
38
File:drivers/sensor/bh1750/bh1750.c
39
Line:144
40
 TRAILING_WHITESPACE: trailing whitespace
41
File:drivers/sensor/bh1750/bh1750.c
[42](https://github.com/zephyrproject-rtos/zephyr/actions/runs/4295302648/jobs/7498324895#step:9:43)
Line:148
[43](https://github.com/zephyrproject-rtos/zephyr/actions/runs/4295302648/jobs/7498324895#step:9:44)
Error: Process completed with exit code 1.
```

**Comment by jeronimoagullo** [comment]
> @jeronimoagullo can you fix the compliance checks:
> 
> ```
> TRAILING_WHITESPACE: trailing whitespace
> 26
> File:drivers/sensor/bh1750/bh1750.c
> 27
> Line:132
> 28
>  SPACING: space required after that ',' (ctx:VxV)
> 29
> File:drivers/sensor/bh1750/bh1750.c
> 30
> Line:138
> 31
>  SPACING: space required after that ',' (ctx:VxV)
> 32
> File:drivers/sensor/bh1750/bh1750.c
> 33
> Line:138
> 34
>  SPACING: space required after that ',' (ctx:VxV)
> 35
> File:drivers/sensor/bh1750/bh1750.c
> 36
> Line:139
> 37
>  SPACING: space required after that ',' (ctx:VxV)
> 38
> File:drivers/sensor/bh1750/bh1750.c
> 39
> Line:144
> 40
>  TRAILING_WHITESPACE: trailing whitespace
> 41
> File:drivers/sensor/bh1750/bh1750.c
> [42](https://github.com/zephyrproject-rtos/zephyr/actions/runs/4295302648/jobs/7498324895#step:9:43)
> Line:148
> [43](https://github.com/zephyrproject-rtos/zephyr/actions/runs/4295302648/jobs/7498324895#step:9:44)
> Error: Process completed with exit code 1.
> ```

Thank you for your comments, it's now done.

File: drivers/sensor/bh1750/Kconfig
Comment (MaureenHelm) [inline drivers/sensor/bh1750/Kconfig:3]: You haven't made any changes to this file; don't add your copyright.

File: drivers/sensor/bh1750/bh1750.c
Comment (MaureenHelm) [inline drivers/sensor/bh1750/bh1750.c:110]: This could benefit from a more meaningful name, something like MEASUREMENT_MODE

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
```
sensor: Rohm bh1750 ambient light sensor

Update Rohm bh1750 ambient light sensor driver.

Signed-off-by: Jeronimo Agullo <jeronimoagullo97@gmail.com>
```
Please explain why you are updating the driver in the commit message.

**Pull request closed without merging**
